### PR TITLE
Added docs_require and fixed RTD build.

### PR DESCRIPTION
**Pull request by nabobalis**
It seems I somehow removed these changes from one of the PRs I already merged.

Fun fact that adding `git+https://github.com/hawkowl/towncrier.git` to setup.cfg and having it parsed by whatever is used by setup.py breaks.

### CODE REVIEW

File: readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+build:
+  image: latest
+
+python:
+  version: 3.6
+  pip_install: true
+  extra_requirements:
+        - all
```
Comment (nabobalis) [inline readthedocs.yml]: I swear it worked without this file. But it seems that I need this at least.

**Comment by Cadair** [comment]
yeah setup.py can't resolve git links. Can we add the dev version of towncrier to the rtd yaml file?

**Comment by nabobalis** [comment]
Looks like thats a no.

**Comment by Cadair** [comment]
Balls. That's kinda a problem, we have needed git versions of the gallery and such before. Maybe we need to go back to requirements for the docs?

**Comment by nabobalis** [comment]
I will see what I can do.

**Comment by nabobalis** [comment]
This seems to work locally, it updates the pip version of town crier and the changelog runs on locally and shows the fancy fragments but for some reason it does not update the changelog on my fork on RTD.

**Pull request closed without merging**